skcosmo/preprocessing/NormalizeScaler.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+import sklearn as sk
+from sklearn.base import TransformerMixin, BaseEstimator
+
+
+class NormalizeScaler(TransformerMixin, BaseEstimator):
+    """A more flexible scaler for features and data, that mimics
+    sklearn.preprocessing.StandardScaler but makes the mean norm of the rows
+    equal to one."""
+
+    def __init__(self, *, with_mean=True, with_std=True, per_feature=False):
+        """Initialize NormalizeScaler. Defines whether it will subtract mean
+        (with_mean=True), apply normalization (with_norm=True) and whether it will
+        normalize each feature separately (per_feature=True)."""
+
+        self.with_mean = with_mean
+        self.with_std = with_std
+        self.per_feature = per_feature
+        self.n_samples_seen_ = 0
+
+    def fit(self, X, y=None):
+        """ Compute mean and scaling to be applied for subsequent normalization. """
+
+        self.n_samples_seen_, self.n_features_ = X.shape
+        self.mean_ = np.zeros(self.n_features_)
+
+        if self.with_mean:
+            self.mean_ = X.mean(axis=0)
+            centred_X = X - self.mean_
+        else:
+            centred_X = X
+
+        self.scale_ = 1.0
+        if self.with_std:
```

Comment:
Why not use `np.linalg.norm` for many of these functions?

Reply:
what for? I mean, I don't really understand how it can simplify something here.